From fc11e9e3746fa9d9004e743188d651bedf7f9772 Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@foss.st.com>
Date: Fri, 7 Mar 2025 11:03:24 +0100
Subject: [PATCH 1663/1695] dt-bindings: add RISAFSUBPROT description for
 reserved memories

Add RISAF subregion macro description in st,protreg property.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 08cb95b9c36e8d91c5924dd9508cb1503387c7ca dt-bindings: add RISAFSUBPROT description for reserved memories]
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@foss.st.com>
Change-Id: I38ff4d4ef4a5561870caeb0733d20d33cac79c70
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/443823
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Tested-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
---
 .../reserved-memory/reserved-memory.yaml      | 46 +++++++++++++++++++
 1 file changed, 46 insertions(+)

diff --git a/documentation/devicetree/bindings/reserved-memory/reserved-memory.yaml b/documentation/devicetree/bindings/reserved-memory/reserved-memory.yaml
index f08b1e04c..080d9ad09 100644
--- a/documentation/devicetree/bindings/reserved-memory/reserved-memory.yaml
+++ b/documentation/devicetree/bindings/reserved-memory/reserved-memory.yaml
@@ -139,6 +139,52 @@ properties:
           RIF_BREN_DIS, region disabled
           RIF_BREN_EN, region enabled
 
+      RISAFSUBPROT(risaf_subregion, dccid, dcen, rden, wren, srcid,
+                   priv, sec, enabled, rlock)
+
+        When a field isn't effective, it should be filled with RIF_UNUSED.
+
+        "risaf_subregion" Contains the ID of the RISAF subregion x to configure.
+          RISAF_SUBREG_ID(x), subregion ID is x.
+
+        "dccid" Contains the CID to which the subregion configuration is
+                delegated. This field has no effect in case of RIF_DCCID_DIS.
+          RIF_CIDx, subregion configuration is delegated to CID x
+
+        "dcen" The subregion configuration is delegated, if set.
+          RIF_DCCID_DIS, subregion delegation disabled
+          RIF_DCCID_EN, subregion delegation enabled
+
+        "rden" Read is enabled in the subregion for the CID defined in the
+               srcid field, if set.
+          RIF_RDEN_DIS, read in subregion disabled
+          RIF_RDEN_EN, read in subregion enabled
+
+        "wren" Write is enabled in the subregion for the CID defined in the
+               srcid field, if set.
+          RIF_WREN_DIS, write in subregion disabled
+          RIF_WREN_EN, write in subregion enabled
+
+        "srcid" Contains the CID which has access to the subregion.
+          RIF_CIDx, CID x has access to the subregion
+
+        "priv" Configures the privilege configuration for this subregion.
+          RIF_PRIV, Only privileged access is granted to the memory subregion
+          RIF_NPRIV, Privileged/Unprivileged access is granted to the memory
+                     subregion
+
+        "sec" Makes the subregion secure, if set.
+          RIF_SEC, subregion is accessible from secure world only
+          RIF_NSEC, subregion is accessible from non-secure world only
+
+        "enabled" Enables the subregion. Access controls applies.
+          RIF_SREN_DIS, subregion disabled
+          RIF_SREN_EN, subregion enabled
+
+        "rlock" The subregion resource is locked, if set.
+          RIF_RLOCK_DIS, subregion resource not locked
+          RIF_RLOCK_EN, subregion resource locked
+
       -RISAB-
       RISABPROT(delegate_en, delegate_cid, sec, default_priv, enabled,
                 cid_read_list, cid_write_list, cid_priv_list)
-- 
2.47.2

